### ISSUE TITLE
Make argument helper more robust 

### ISSUE THREAD

**Issue by voidcycles** [uic-evl/omicron#24]
The libconfig::ArgumentHelper interface is extremely fragile and error prone.  It silently ignores and consumes unrecognised command line options.  Please just use boost::program_options internally.

**Comment by febret** [comment]
I agree that boost::program_options would be better, but ArgumentHelper is exposed to third party apps trhough the oargs() call on orun, so it can't be replaced easily without breaking backward compatibility. My suggestion would be to fix specific issues in ArgumentHelper (source code in inclued in omicron) instead of substituting it.
